snuba_sdk/metrics_visitors.py

```diff
@@ -148,6 +208,18 @@ def visit(self, metric: Metric) -> Mapping[str, str]:
         }
 
 
+class MetricMQLPrinter(MetricVisitor[Mapping[str, str]]):
+    def visit(self, metric: Metric) -> Mapping[str, str]:
```

Comment:
This requires more thought. We need the id in the MQL query, at least until we have the indexer working in Snuba. Otherwise Snuba won't be able to actually run the query.

Reply:
I agree, but I don't think we should encode the ID into the MQL query as a temporary solution, since we can already query snuba with the new metrics layer as it exists right now. I think moving the indexer to Snuba should be a requirement in order for the new snuba endpoint to function.

Reply:
I don't want to block work on this layer behind the indexer work. I would rather build this in a way that supports IDs, then remove that support once the indexer work is done.

We don't know how long that will take or how easy it will be, and blocking metrics layer development with it I think will end up hampering us in the future.

What we should do is call this out as tech debt that we need to clean up once the indexer work is done.

Reply:
Sure that makes sense. Let's chat about how we might want to do this offline, because I think there's an opportunity to group this data together with the other context fields (scope, limit, totals, start, end, etc.) cleanly.